File: python/CaboCha/cabocha_example.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import CaboCha
cabocha = CaboCha.Parser('--charset=UTF8')
#sent = "太郎はこの本を二郎を見た女性に渡した。".encode('utf-8')
sent = "太郎はこの本を二郎を見た女性に渡した。"

print("print sent")
print(sent)

print("print cabocha.parseToString(sent)")
print(cabocha.parseToString(sent))

tree = cabocha.parse(sent)

print("print tree.toString(CaboCha.FORMAT_LATTICE)")
print(tree.toString(CaboCha.FORMAT_LATTICE))

print("print tree.toString(CaboCha.FORMAT_XML)")
print(tree.toString(CaboCha.FORMAT_XML))

print("print tree.toString(0)")
print(tree.toString(0))
print("print tree.toString(1)")
print(tree.toString(1))
print("print tree.toString(2)")
print(tree.toString(2))
print("print tree.toString(3)")
print(tree.toString(3))
print("print tree.toString(4)")
print(tree.toString(4))
print("print tree.toString(5)")
print(tree.toString(5))

File: ruby/CaboCha/cabocha_example.rb
# -*- coding: utf-8 -*-

require 'CaboCha'

cabocha = CaboCha::Parser.new('--charset=UTF8')

#sent = "サッカーワールドカップで、日本はデンマークに快勝した。"
sent = "太郎はこの本を二郎を見た女性に渡した。"

puts "puts sent"
puts sent

puts "puts cabocha.parseToString(sent)"
puts cabocha.parseToString(sent)

tree = cabocha.parse(sent)
puts "puts tree.toString(0)"
puts tree.toString(0)
puts "puts tree.toString(1)"
puts tree.toString(1)
puts "puts tree.toString(2)"
puts tree.toString(2)
puts "puts tree.toString(3)"
puts tree.toString(3)
puts "puts tree.toString(4)"
puts tree.toString(4)
puts "puts tree.toString(5)"
puts tree.toString(5)

